presto-orc/src/main/java/com/facebook/presto/orc/stream/ByteOutputStream.java

```diff
@@ -107,18 +108,17 @@ public void writeByte(byte value)
 
     private void flushSequence()
     {
-        if (size == 0) {
-            return;
-        }
-
         if (runCount >= MIN_REPEAT_SIZE) {
             buffer.writeByte(runCount - MIN_REPEAT_SIZE);
             buffer.writeByte(lastValue);
         }
         else {
             buffer.writeByte(-size);
-            for (int i = 0; i < size; i++) {
-                buffer.writeByte(sequenceBuffer[i]);
+            if (size == 1) {
```

Comment:
`buffer.writeBytes(byte [] arr)` call is quite expensive, writing a single byte is cheap.

Through some debugging I found that there are a lot of cases when we are only flushing a single byte before switching to RLE.  So I put and tested a small optimization to decide whether to write a single value or an array.